python_modules/dagster/dagster/core/definitions/asset_layer.py

```diff
@@ -59,6 +61,29 @@ def __new__(
         )
 
 
+def _resolve_input_to_destinations(
+    name: str, node_def: NodeDefinition, handle: NodeHandle
+) -> Sequence[NodeInputHandle]:
+    """
+    Recursively follow input mappings to find all op inputs for a graph input.
+    """
+    if not isinstance(node_def, GraphDefinition):
+        # must be in the op definition
+        return [NodeInputHandle(node_handle=handle, input_name=name)]
+    all_destinations: List[NodeInputHandle] = []
+    for mapping in node_def.input_mappings:
+        if mapping.definition.name != name:
+            continue
+        # recurse into graph structure
+        all_destinations += _resolve_input_to_destinations(
```

Comment:
Not a huge deal, because in most cases N will be small, but kind of unperformant to do all this list concatenation. An O(N) way of doing this could be to use `yield` and `yield from`.